src/csv_processor.cpp

```diff
@@ -1,25 +1,268 @@
+  }
+
+  // We cannot assume the order of columns. From the Fivetran Partner SDK docs:
+  // "Always read the CSV file header to determine the column order."
+  // Therefore, we do not set the `columns` parameter, but `column_types` which
+  // is a mapping from a column name to a type:
+  // column_types={'colB':'VARCHAR','colA':'INTEGER'}.
+  // DuckDB detects the order of columns by reading the header row.
+  // If no columns are specified, DuckDB will auto-detect all column types.
+  query << ", column_types={";
+  for (const auto &column : columns) {
+    // Even if we do not specify the type for this column, DuckDB will figure it
+    // out itself because of auto_detect=true
+    if (column.type == duckdb::LogicalTypeId::INVALID) {
+      continue;
+    }
+
+    query << duckdb::KeywordHelper::WriteQuoted(column.name, '\'') << ":";
+
+    query << "'" << duckdb::EnumUtil::ToString(column.type);
+    if (column.type == duckdb::LogicalTypeId::DECIMAL && column.width > 0) {
+      query << "(" << std::to_string(column.width) << ","
+            << std::to_string(column.scale) + ")";
+    }
+    // DuckDB can handle trailing comma
+    query << "',";
+  }
+  query << "}";
+}
+
+/// Generates a DuckDB SQL query string to read a CSV file with the specified
+/// properties
+std::string generate_read_csv_query(const std::string &filepath,
+                                    const IngestProperties &props,
+                                    const CompressionType compression) {
+  std::ostringstream query;
+  query << "FROM read_csv("
+        << duckdb::KeywordHelper::WriteQuoted(filepath, '\'');
+  // We set auto_detect=true so that DuckDB can detect the dialect options that
+  // we do not set explicitly. It further helps with detecting column types if
+  // there happen to be columns whose type we did not set explicitly. This is
+  // not expected to happen, but is more robust this way.
+  query << ", auto_detect=true";
+  query << ", delim=','";
+  query << ", encoding='utf-8'";
+  // Escaped string in CSV looks like this: "A ""quoted"" word"
+  query << ", escape='\"'";
+  query << ", header=true";
+  query << ", new_line='\\n'";
+  query << ", quote='\"'";
+  // We do not specify timestampformat, see below.
+  // Date format: 2025-12-31
+  query << ", dateformat='%Y-%m-%d'";
+  if (!props.null_value.empty()) {
+    query << ", nullstr="
+          << duckdb::KeywordHelper::WriteQuoted(props.null_value, '\'');
+    query << ", allow_quoted_nulls=true";
+  }
+  // The default max_line_size is 2MB. The default buffer size is
+  // 16*max_line_size=32MB. We only set this option to a higher value if
+  // requested.
+  constexpr std::uint32_t default_csv_block_size_mb = 32;
+  if (props.csv_block_size_mb > default_csv_block_size_mb) {
+    const auto max_line_size = props.csv_block_size_mb * 1000 * 1000 / 16;
+    query << ", max_line_size=" << std::to_string(max_line_size);
+    query << ", buffer_size=" << std::to_string(max_line_size * 16);
```

Comment:
max_line_size is calculated from buffer size anyway, right? No need to divide by 16 and then multiply by the same